pkg/products/threescale/reconciler.go

```diff
@@ -1444,3 +1457,42 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8s
+		cl.Spec.ApplicationMenu.Section = "Red Hat Applications"
+		cl.Spec.Href = fmt.Sprintf("%v/auth/rhsso/bounce", r.Config.GetHost())
+		cl.Spec.Location = consolev1.ApplicationMenu
+		cl.Spec.Text = "API Management"
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("error creating or updating 3Scale console link, %s", err)
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) deleteConsoleLink(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
```

Comment:
I wonder could we possibly remove the console link the same way as it's done with solution explorer? https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/solutionexplorer/reconciler.go#L156 - in my understanding it would save us creating a new function and keep the codebase slightly shorter?

Reply:
Personally I don't agree with the style taken in the solution explorer reconciler, I do understand the reasoning behind it. 

With in the 3 scale reconciler the style seem to be have a function/method for a body of work been done (for the most part). To my it would be better to keep the style of code when the changes are been made.